src/historywork/DownloadBucketsWork.cpp

```diff
@@ -68,6 +69,10 @@ DownloadBucketsWork::yieldMoreWork()
     auto hash = *mNextBucketIter;
     FileTransferInfo ft(mDownloadDir, HISTORY_FILE_TYPE_BUCKET, hash);
     auto w1 = std::make_shared<GetAndUnzipRemoteFileWork>(mApp, ft, mArchive);
+    if (!mArchive)
+    {
+        mDownloads[hash] = w1;
```

Comment:
What do you think about logging in `selectRandomReadableHistoryArchive` (Change the "Fetching" message to INFO) instead of keeping track of the archives here? You wouldn't have to maintain this map and it would expose the chosen archive for every work.

Reply:
I picked the current approach because it seems more operator-friendly: we try to provide as much useful information as possible in a single log message when something bad actually happened (otherwise, an operator needs to know what to look for in multiple places).